editor/scene/3d/camera_3d_editor_plugin.cpp

```diff
@@ -79,21 +83,52 @@ Camera3DEditor::Camera3DEditor() {
 	preview->connect(SceneStringName(pressed), callable_mp(this, &Camera3DEditor::_pressed));
 }
 
+bool Camera3DPreview::camera_preview_folded = false;
+
 void Camera3DPreview::_update_sub_viewport_size() {
-	sub_viewport->set_size(Node3DEditor::get_camera_viewport_size(camera));
+	if (sub_viewport != nullptr) {
+		const Size2i camera_size = Node3DEditor::get_camera_viewport_size(camera);
+		centering_container->set_ratio(camera_size.aspect());
+	}
 }
 
-Camera3DPreview::Camera3DPreview(Camera3D *p_camera) :
-		TexturePreview(nullptr, false), camera(p_camera), sub_viewport(memnew(SubViewport)) {
-	RenderingServer::get_singleton()->viewport_attach_camera(sub_viewport->get_viewport_rid(), camera->get_camera());
-	add_child(sub_viewport);
+void Camera3DPreview::_camera_exiting() {
+	if (sub_viewport != nullptr) {
+		const Size2i camera_size = Node3DEditor::get_camera_viewport_size(camera);
+		centering_container->set_ratio(camera_size.aspect());
+	}
+}
+
+void Camera3DPreview::_toggle_folding(bool p_folded) {
+	camera_preview_folded = p_folded;
+}
 
-	TextureRect *display = get_texture_display();
-	display->set_texture(sub_viewport->get_texture());
-	sub_viewport->connect("size_changed", callable_mp((CanvasItem *)display, &CanvasItem::queue_redraw));
-	sub_viewport->get_texture()->connect_changed(callable_mp((TexturePreview *)this, &Camera3DPreview::_update_texture_display_ratio));
+Camera3DPreview::Camera3DPreview(Camera3D *p_camera) {
+	camera = p_camera;
+	camera->connect(SceneStringName(tree_exiting), callable_mp(this, &Camera3DPreview::_camera_exiting));
+
+	FoldableContainer *folder = memnew(FoldableContainer);
+	folder->set_title(TTRC("Camera Preview"));
+	folder->set_title_text_overrun_behavior(TextServer::OVERRUN_TRIM_ELLIPSIS);
+	folder->set_folded(camera_preview_folded);
+	folder->connect(SNAME("folding_changed"), callable_mp(this, &Camera3DPreview::_toggle_folding));
```

Comment:
Unless this is created repeatedly using `SNAME` here is unnecessary, it increases binary size and should only be used in hot paths generally

Reply:
It creates a local static variable, which takes memory, and adds to binary size AFAIK, but also see the note:
https://github.com/godotengine/godot/blob/63e14e13f972671a43abcf019962129e5cca7d01/core/string/string_name.h#L200-L210

Reply:
Oh, great. Thanks!